docs/guides/securing-apps/token-exchange.adoc

```diff
@@ -6,6 +6,322 @@ title="Using token exchange"
 priority=120
 summary="Configuring and using Token exchange with {project_name}">
 
+Token exchange is the process, which allows client application to exchange one token for another token. In {project_name}, there are two features, which implement token exchange:
+
+* <<_standard-token-exchange,Standard token exchange: version 2 (V2)>> - This is fully supported token exchange implementation, which is enabled by default once {project_name} server is started.
+* <<_legacy-token-exchange,Legacy token exchange: version 1 (V1)>> - This is preview feature, which is not enabled by default once {project_name} server is started.
+
+Here is a short summary of the current capabilities of {project_name} around token exchange.
+
+. A client can exchange an existing {project_name} token created for a specific client for a new token targeted to a different client in the same realm
+. A client can exchange an existing {project_name} token for an external token, i.e. a linked Facebook account
+. A client can exchange an external token for a {project_name} token.
+. A client can impersonate a user
+
+The standard token exchange supports only the use-case (1), when legacy token exchange provides also other three use-cases. When possible, it is highly recommended to use standard token exchange V2 as that
+one is supported and will be maintained for the future. Using legacy token exchange is possible if you need some of the last three use cases, but note that legacy token exchange is preview and may not be
+backwards compatible with the future {project_name} versions. It is possible to enable both token exchange features and use them together, which is useful if you need to use both internal-internal exchange
+provided by V2 together with other use cases, which are provided only by V1. The summary comparison of standard and legacy token exchange can be seen in <<_standard-token-exchange-comparison,this section>>.
```

Comment:
```suggestion
The standard token exchange supports only use-case (1). The legacy token exchange support the four use-cases, but it is a preview feature. Therefore, the standard token exchange V2 is recommended  since it is supported and will be maintained for the future. The legacy token exchange is useful for last three use cases, but it may not be
backwards compatible with the future {project_name} versions. You can also enable both token exchange features and use them together. For example, you could use both internal-internal exchange
provided by V2 together with other use cases that are supported by V1. For more details, see this <<_standard-token-exchange-comparison,token exchange comparison>>.
```